### PR TITLE
docs(cognito): fix up code samples so that they compile

### LINKED ISSUE #9185: Reference examples for C#/.NET show "new Struct" when it should be "new <module>Props"
https://docs.aws.amazon.com/cdk/api/latest/dotnet/api/Amazon.CDK.AWS.Cognito.html

Example of creating a user pool:
```
new UserPool(this, "myuserpool", new Struct {
    UserPoolName = "myawesomeapp-userpool"
});
```

This should be:
```
new UserPool(this, "myuserpool", new UserPoolProps {
    UserPoolName = "myawesomeapp-userpool"
});
```

This problem is all over the place in the .NET reference examples. Very annoying.

This is a 📕 documentation issue

### PR DESCRIPTION
**Pull request by shivlaks**
Added a default fixture with the imports needed to compile snippets.
Fixed up a few parameter names, required parameters, references that were incorrect.

Verified that these snippets (with fixture) compile by running `compile-samples` in
the `scripts` directory.

Closes #9185

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by eladb — APPROVED** [review]

File: packages/@aws-cdk/aws-cognito/README.md [hunk 1]
```diff
@@ -211,11 +211,11 @@ Additionally, the property `enableSmsRole` can be used to override the CDK's def
...
 ```ts
-import { Role } from '@aws-cdk/aws-iam';
-
-const poolSmsRole = new Role(this, 'userpoolsmsrole', { /* ... */ });
+const poolSmsRole = new iam.Role(this, 'userpoolsmsrole', {
+  assumedBy: new iam.ServicePrincipal('foo'),
+});
 
-new UserPool(this, 'myuserpool', {
+new cognito.UserPool(this, 'myuserpool', {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/README.md:218]: Not a big fan of prefixing `cognito` everywhere. Would've preferred if the fixture had individually imported what's required, so the README stays clean.

OTOH doing so would mean that any new type being used here would have the fixture be updated. Given that we have no validation for this, maybe this is ok.

File: packages/@aws-cdk/aws-cognito/README.md [hunk 2]
```diff
@@ -320,22 +320,24 @@ Lambda triggers can either be specified as part of the `UserPool` initialization
...
+  lambdaTriggers: {
     createAuthChallenge: authChallengeFn,
     // ...
   }
 });
 
-userpool.addTrigger(UserPoolOperation.USER_MIGRATION, new lambda.Function(this, 'userMigrationFn', {
-  // ...
+userpool.addTrigger(cognito.UserPoolOperation.USER_MIGRATION, new lambda.Function(this, 'userMigrationFn', {
+    runtime: lambda.Runtime.NODEJS_10_X,
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/README.md:338]: fix indent

File: packages/@aws-cdk/aws-cognito/README.md [hunk 3]
```diff
@@ -320,22 +320,24 @@ Lambda triggers can either be specified as part of the `UserPool` initialization
...
 
 ```ts
-import * as lambda from '@aws-cdk/aws-lambda';
-
 const authChallengeFn = new lambda.Function(this, 'authChallengeFn', {
-  // ...
+  runtime: lambda.Runtime.NODEJS_10_X,
+  handler: 'index.handler',
+  code: lambda.Code.fromInline('auth challenge'),
 });
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/README.md:327]: Is this necessary? I thought rosetta translations work as long as it was valid typescript. Does it have to actually compile and work?

I would've preferred to leave this open for the user to define and rosetta translation snippet does the same.

File: packages/@aws-cdk/aws-cognito/README.md [hunk 4]
```diff
@@ -521,15 +525,17 @@ owned, and whose certificate is registered in AWS Certificate Manager.
...
+const pool = new cognito.UserPool(this, 'Pool');
 
 pool.addDomain('CognitoDomain', {
   cognitoDomain: {
     domainPrefix: 'my-awesome-app',
   },
 });
 
-const domainCert = new acm.Certificate.fromCertificateArn(this, 'domainCert', certificateArn);
+const certificateArn = 'arn:aws:acm:us-east-1:123456789012:certificate/11-3336f1-44483d-adc7-9cd375c5169d';
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/README.md:536]: Why not move this into the fixture?

File: packages/@aws-cdk/aws-cognito/README.md [hunk 5]
```diff
@@ -398,14 +398,18 @@ The following code shows how different attributes provided by 'Login With Amazon
...
 ```ts
-new UserPoolIdentityProviderAmazon(stack, 'Amazon', {
-  // ...
+const userpool = new cognito.UserPool(this, 'Pool');
+
+new cognito.UserPoolIdentityProviderAmazon(this, 'Amazon', {
+  clientId: 'amzn-client-id',
+  clientSecret: 'amzn-client-secret',
+  userPool: userpool,
   attributeMapping: {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/README.md:407]: Not a fan of this change either. Now the focus is diffused and doesn't focus on just the `attributeMapping` property which this section is talking about.

How about defining the whole set of properties in the rosetta fixture and doing something like

```ts
new UserPoolIdentityProviderAmazon(stack, 'Amazon', {
  ...otherIdentityProviderAttrs, // this is defined in the fixture
  attributeMapping: {
  }
});
```

**Pull request merged**